engine/src/main/java/org/terasology/engine/world/chunks/pipeline/InitialChunkProvider.java

```diff
@@ -0,0 +1,23 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.world.chunks.pipeline;
+
+import org.joml.Vector3ic;
+import org.terasology.engine.world.chunks.Chunk;
+
+import java.util.Set;
+
+/**
+ * An InitialChunkProvider provides input chunks to the ChunkProcessingPipeline.
+ * All methods should be able to be called by multiple threads simultaneously, taking care of any necessary synchronization themselves.
+ */
+public interface InitialChunkProvider {
+    boolean hasNext();
+
+    /**
+     * @param currentlyGenerating the set of chunks which are currently being processed.
+     *  This lets the InitialChunkProvider discard those chunks before trying to generate them at all.
+     */
+    Chunk next(Set<Vector3ic> currentlyGenerating);
```

Comment:
The parameter on `next` is so that it can check if a chunk is already processing before generating it; it doesn't really matter for `hasNext`, but because they're different `hasNext` can return true but then `next` return null. It might be better to either just have one method which can return `null` if there are no new chunks, give both methods the same parameter, or make `next` return a `Producer<Chunk>` or something.

Reply:
Okay, I just removed `hasNext` and made `next` return an `Optional`, which is simpler.